packages/synapse-react-client/src/utils/functions/TableColumnSchemaUtils.ts

```diff
@@ -4,40 +4,67 @@ import {
   TableUpdateTransactionRequest,
 } from '@sage-bionetworks/synapse-types'
 import { SetOptional } from 'type-fest'
-import { isEqual } from 'lodash-es'
 import { createColumnModels } from '../../synapse-client'
+import { isEqualTreatUndefinedAsMissing } from './IsEqualTreatUndefinedAsMissing'
+
+function validateProposedSchema(
+  oldSchema: ColumnModel[],
+  proposedSchema: SetOptional<ColumnModel, 'id'>[],
+): void {
+  // If the proposed schema contains any IDs that are not in the old schema, throw an error
+  // We do this so we can accurately track updates to existing columns by using the ID that is already present
+  const oldColumnModelIds: Set<string> = new Set()
+  for (const columnModel of oldSchema) {
+    oldColumnModelIds.add(columnModel.id)
+  }
+  for (const columnModel of proposedSchema) {
+    if (columnModel.id != null && !oldColumnModelIds.has(columnModel.id)) {
+      throw new Error(
+        `Proposed schema contains a new column model with ID ${columnModel.id} that is not in the old schema.`,
+      )
+    }
+  }
```

Comment:
At the layers above this, we remove IDs from new columns, but add this validation check to prevent regression in the future